### PR TITLE
Split Header out into per page components

### PR DESCRIPTION
**Pull request by RandomAshes**
## Issue
Resolves N/A

## What was done
Before, the Header component had a bunch of conditionals in it so it would hide and show various things based on page. 
In this PR, I turned the main Header.tsx component into a template, and then used that template with different props, or created a wrapper component per page so that each page can handle its own needs.

Screen Shot 2022-08-04 at 5 03 41 PM

## How to test
1. Ensure the header behaves the same as it did before on all pages.

**Expected Behavior:**

_Landing page_:
* should not have a header at all

_Dashboard Page:_ 
* should show header
* Peoplemover logo should be static (not a link)
* should only show sign out button in account dropdown
* should NOT show space name or anything else except logo and account dropdown
* 
_Contact Us Page:_ 
* should show header
* Peoplemover logo should link back to the dashboard page
* should only show sign out button in account dropdown
* should NOT show space name or anything else except logo and account dropdown

_Space Page:_
* should show header
* Peoplemover logo should link back to the dashboard
* should show all 3 buttons in the account dropdown
* should show space name
* should show "Time On Product" link

_Time on product Page:_
* should show header
* Peoplemover logo should link back to the dashboard
* should show all 3 buttons in the account dropdown
* should show space name
* should show "Back" link

_Error Pages_
* should show header
* Peoplemover logo should link back to the dashboard
* should not show account dropdown or anything but the logo

### CODE REVIEW

File: ui/src/TimeOnProductPage/TimeOnProductHeader/TimeOnProductHeader.test.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        enableInviteUsersToSpace();
+
+        ({container} = renderWithRecoil(
+            <MemoryRouter initialEntries={[`/${TestData.space.uuid}/timeonproduct`]}>
+                <TimeOnProductHeader />
+            </MemoryRouter>,
+            ({set}: MutableSnapshot) => {
+                set(CurrentSpaceState,  TestData.space)
+            }
+        ));
```
Comment (hirava271) [inline ui/src/TimeOnProductPage/TimeOnProductHeader/TimeOnProductHeader.test.tsx:47]: Just for my own learnings, could you explain me why do we need to render component differently here?

File: ui/src/Common/Header/Header.test.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    shouldShowAllAccountDropdownOptions,
+} from 'Utils/HeaderTestUtils';
+import {MutableSnapshot} from 'recoil';
+import {CurrentUserState} from 'State/CurrentUserState';
+
+const debounceTimeToWait = 100;
+
+describe('Header', () => {
+    beforeEach(() => {
+        window.runConfig = {invite_users_to_space_enabled: true} as RunConfig;
```
Comment (hirava271) [inline ui/src/Common/Header/Header.test.tsx:41]: can we utilize that enableUsersToSpace function from utils here?

File: ui/src/TimeOnProductPage/TimeOnProductHeader/TimeOnProductHeader.test.tsx (hunk 1 above)
Comment (RandomAshes) [reply ui/src/TimeOnProductPage/TimeOnProductHeader/TimeOnProductHeader.test.tsx:47]: Differently than what?

File: ui/src/Common/Header/Header.test.tsx (hunk 2 above)
Comment (RandomAshes) [reply ui/src/Common/Header/Header.test.tsx:41]: Yes! Good catch

**Review by hirava271 — APPROVED** [review]

**Pull request merged**